### ISSUE TITLE
Firefox always showing scheduling rule as "must" on upgrade/clone

### ISSUE THREAD

**Issue by nhumrich** [rancher/rancher#8494]
**Rancher Versions:**
Server: 1.5.3
scheduler: 0.7.5

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Create a service with a schedule rule, for example "host should not have a service with the name {service name}. Click upgrade. Once upgrade is complete, click approve. Then go into the upgrade menu again. Click into the schedule rules, and the rule you just added does not show up. Some rule shows up, but its not the right one.

**Results:** 
the rule you just added does not show up. Some rule shows up, but its not the right one. I havent tested it, but this could potentially override an existing scheduling rule on accident.

**Comment by aemneina** [comment]
@nhumrich is that the exact use case you ran through? I tried this on rancher v1.5.5 + scheduler v0.7.5 and wasn't able to reproduce. Can you try it on 1.5.5?

**Comment by nhumrich** [comment]
Just tried it on 1.5.5, no difference:
If I go look at the docker-compose yaml from rancher I get the following:

```
version: '2'
services:
  java-all-projects:
    image: canopytax/java-service:xxxxxx
    labels:
      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=java-all-projects
      service: java-all
```

But when I go into the UI it shows:

![selection_069](https://cloud.githubusercontent.com/assets/4661889/25157039/f9c1a04a-245b-11e7-9958-f342d8362f39.png)

The scheduling rule is incorrect.

**Comment by vincent99** [comment]
Are you doing global scale?  Soft affinity ("should") are not allowed with global.

**Comment by oskapt** [comment]
I can duplicate this. my workflow is as follows:
* create service with scale of 2
* add the following scheduling rules:
  * `host must have a host label of storage=true`
  * `host must not have a service with the name storage/gluster`
* after creating the service, when i return to the scheduling screen, i see the attached image 1
  * `host must not have a service...` is displayed incorrectly
* when i click the 'Condition' or 'Field' dropdowns in the first (incorrect) row, I see that the correct item is _selected_ (images 2 and 3); it just isn't displayed. However, after viewing it, the incorrect value is now selected and will probably be saved.

I was also seeing this in 1.5.3 and upgraded to 1.5.5 to verify that it's still an issue. An interesting note is that I had these two rules in the 1.5.3 installation, and after upgrading Rancher (not yet upgrading the service), I only had the incorrectly-displayed 'must not have a service' rule. The host label rule had vanished completely.

![ui_sched_1](https://cloud.githubusercontent.com/assets/171292/25180052/d9b37ad0-24e2-11e7-81ef-41ce75044ea4.png)
![ui_sched_3](https://cloud.githubusercontent.com/assets/171292/25180053/d9b744f8-24e2-11e7-89e9-ae3b329cab6e.png)
![ui_sched_2](https://cloud.githubusercontent.com/assets/171292/25180051/d9a825c2-24e2-11e7-968c-c9360b9a7904.png)

**Comment by nhumrich** [comment]
@vincent99 no, I am not using global scale. Just scale of 2

**Comment by tfiduccia** [comment]
@nhumrich I am unable to reproduce your problem.  In my yaml I added two rules:
```
labels:
    io.rancher.scheduler.affinity:host_label: storage=true
    io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=storage/gluster
```

When I upgrade/edit the service I see this:
![image](https://cloud.githubusercontent.com/assets/11514927/25296120/882e559e-269b-11e7-83f0-facb48092a92.png)

Can you give me the label section of your yaml, to se if maybe there is something else causing the problem?

**Comment by nhumrich** [comment]
@tfiduccia I included the label section of my yaml

**Comment by nhumrich** [comment]
Possibly relevant: I am on firefox

**Comment by tfiduccia** [comment]
@nhumrich - 

Using your labels section, that problem is indeed just an issue with the UI in firefox.  "Should have" is actually being set and the config and all the info will work properly.  But in Firefox when you go to upgrade/clone, it 'looks' as though 'must' is being selected.  If you click on the drop-down you can actually see that 'should not' is selected.  Also if you use Chrome, you won't see this at all.

So for firefox, the drop-down is always showing reverting back to "must" on edit.  I did do a little testing and as long as you don't actually use the drop-down, it will keep your original rule and work fine.

**Comment by phal0r** [comment]
@tfiduccia @vincent99 
I can confirm this Bug on a recent Firefox Version and Rancher 1.6. Our Rule is `MUST NOT` `have a service with name` and the value is the selected service. On Chrome everything is working as expected, Firefox shows `MUST` `have a host label` and still the service name as value, which makes no sense at all. This is still a UI Bug, which can confuse people, isn't it?

**Comment by vincent99** [comment]
This was fixed 2 months ago in 1.6.11:
https://github.com/rancher/ui/commit/98fdb7e8f6656305006e3a871820aad3f9f99c69
